### DISCUSSION TITLE
Cannot get github enterprise pull requests to work

### DISCUSSION THREAD

**Discussion by fipp** [WeblateOrg/weblate#7687] [Q&A] [answered]
Hi! I've set up a weblate instance based on the `4.12.1-1` docker image. The repository I'm connecting with is on a company `GitHub Enterprise Server 3.3.1` instance.

A regular `Git` setup for my component works fine, commits are pushed to the designated branch without problem.

But when I switch to pull request, I see this error in the log when I try to push:

> `git: failure ERROR: Permission to my-github-username/my-repo.git denied to deploy key`

I don't understand why. The `WEBLATE_GITHUB_USERNAME` environment variable is set to `my-github-username` (my github enterprise username), and the `WEBLATE_GITHUB_TOKEN` is a personal access token generated with the following rights:

<img width="744" alt="image" src="https://user-images.githubusercontent.com/1716906/169984592-761a21e6-d370-4057-bb3f-327c75f13be7.png">

I have tried using the personal access token on the terminal with `git clone https://accesstoken@github.mycompany.io/myorg/my-repo.git` with success (also push works, when cloning without token in url, I get asked for credentials).

The weblate ssh public key is also added as a deploy key to the repo in github.

Questions:
* When using github pull request, should the "Source Code Repository" url be ssh (`git@github.mycompany.io:myorg/my-repo.git`) or https (`https://github.mycompany.io/myorg/my-repo.git`)? 
    * If https, should the personal access token be part of the url? 
* If I set "Repository Branch", the pull request target will be that branch, correct?

<img width="461" alt="image" src="https://user-images.githubusercontent.com/1716906/169985667-a06fe0d4-9dcb-4d4f-8ca1-af28aeb638a1.png">

**Comment by fipp** [comment]
Update. It seems like there is a difference when the repo is owned by a github organization, versus myself.

I created `github.mycompany.io/my-github-username/test` and it worked, the pull request was created. But with `github.mycompany.io/myorg/test`, it did not work, I get the `denied to deploy key` error. I find this odd, as I'm a member of that github organization 🤔

**Comment by nijel** [comment]
Seems that failing thing is git access via SSH, not an API access to create the pull request. Try using https URL for the repository (including the access token).

**Comment by fipp** [comment, marked as answer]
> Seems that failing thing is git access via SSH, not an API access to create the pull request. Try using https URL for the repository (including the access token).

I tried now with https url instead of ssh url, and the issue was the same.

But I think I figured it out now. The problem was that the ssh public key was added as a deploy key to the repo, and not as a ssh key to my account. My theory is that this is important when the repository is not owned by my account, but by the github organization I'm a part of.
